crates/runtime/src/instance/allocator/pooling.rs

```diff
@@ -452,6 +482,19 @@ impl InstancePool {
             let mut memory = mem::take(memory);
             debug_assert!(memory.is_static());
 
+            #[cfg(feature = "memfd-allocator")]
+            {
+                if let Memory::Static {
+                    memfd_slot: Some(memfd_slot),
+                    ..
+                } = memory
+                {
+                    let memfd_slot = unsafe { memfd_slot.as_mut().unwrap() };
+                    let _ = memfd_slot.clear_and_remain_ready();
+                    continue;
+                }
+            }
```

Comment:
Well to some extent I don't really think this needs a major refactoring. While ideally we have `MemoryKind` variants or something like that what we have today in this PR is two `Memory` variants where one has fields which configures uffd or memfd if they're enabled. I think that this logic in this particular loop can still be structured as "here's an arm for each case" as opposed to "memfd is special and gets the early-`continue`". I was originally confused reading this loop because the loop always ends with `madvise`  and it sort of seems like memfd doesn't end with `madvise` when it in fact does.

Reply:
Ah, right, this wasn't nearly as much a refactor as I thought; sorry! Done.